OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseUnitMobileNotSupportedFragment.kt

```diff
@@ -247,24 +249,28 @@ class CourseUnitMobileNotSupportedFragment : CourseUnitFragment() {
     override fun onResume() {
         super.onResume()
         unit?.let {
+            val isUpgradeDeadlinePassed = getBooleanArgument(EXTRA_IS_UPGRADE_DEADLINE_PASSED, true)
             if (it.authorizationDenialReason == AuthorizationDenialReason.FEATURE_BASED_ENROLLMENTS
-                && environment.appFeaturesPrefs.isValuePropEnabled()
+                && environment.appFeaturesPrefs.isValuePropEnabled() && !isUpgradeDeadlinePassed
             ) {
                 environment.analyticsRegistry.trackLockedContentTapped(it.courseId, it.blockId)
             }
         }
     }
 
     companion object {
+        private const val EXTRA_IS_UPGRADE_DEADLINE_PASSED = "is_upgrade_deadline_passed"
```

Comment:
as `EXTRA_IS_SELF_PACED ` is declared in Router, better to move `EXTRA_IS_UPGRADE_DEADLINE_PASSED ` too.